### ISSUE TITLE
Implement a visual composition guide overlay system for camera layer

### ISSUE THREAD

**Issue by Jose-Moreno** [pencil2d/pencil#1082]
### Issue Summary

Having visual composition guides is a very important feature for animation planning of what happens in your picture plane (a.k.a what you see) Having an overlay feature (that is not renderable on export of course) will help for the cinematography aspect of the animation when creating storyboards, layouts and organizing how the elements are supposed to be seen (this is a feature to aid visual design in animation basically). 

As for overlay (OVL) elements we could have a few elements to enable like:

- [x] Center Point OVL
- [x] Visual Thirds OVL (A grid-like division that has vertical and horizontal guides that divide the camera space in thirds)
- [x] Golden Ratio OVL (this is used...sometimes, but could be replaced with a more functional overlay)
- [x] Title Safe Area / Box OVL
- [x] Action Safe Area / Box OVL Usually some people use 5% For the "Action safe" and 10% for the "Title safe" areas. This comes from the era of 4:3 ratio screens and when cathodic ray "tubes" cut part of the image, nowadays this is still handled for flat screens even if there's not actual image loss but there are some countries that have regulations for those numbers and as such should be changeable.

There could be more overlays, and I was thinking that similarly with Palettes being XML files we could also have these overlays as scriptable drawings via XML or something, like if you create the designs via SVG tags in the XML OVL file, that would allow anyone to add more than the minimum overlays we ship, without us having to maintain it. So people could create their designs on inkscape or illustrator, export to svg, and then we use the xml info in our "composition guides" file

Edit: For understanding what compositon guides are and which ones could be implemented (as they are actually used) please [refer to this article](https://fstoppers.com/architecture/ultimate-guide-composition-part-one-just-say-nokeh-31359)

**Note**: It's possible that you could use the current grid system (with the independent x & y values) to mimic some of these overlays, but often times it's better to have dedicated elements for these kind of things, particularly for the title and action safe areas; Also it's important to mention that you normally want them to stack one on top of each other, rather than only show each of them at a time.

**Comment by davidlamhauge** [comment]
I've assigned myself to this @Jose-Moreno . Let's see if I can pull it off.
You suggest that we should use an XML file for storing the designs. I'm not sure that would work too good. As it is now, everyone can import an overlay and use it as they like. 
At least I think the standard overlays you mention should be done in code. The user should be able to choose the preferred values for Text safe and Action safe percentages, but the others are as is.
What do you think?

**Comment by Jose-Moreno** [comment]
@davidlamhauge I always think of scalability and the possibility of not depending on the developers to add custom elements that can be different in various environments, but since this has so little interest, I don't mind if we ship a minimum set of overlays created via code as you suggest.

The xml idea was more thinking about compositional guides like the rule of thirds, the golden spiral and others that filmakers & painters use, but that can be added later anyway.

**Comment by Jose-Moreno** [comment]
Implemented by #1301 | Closing

I feel the implementation works quite well from an intuitive point of view, and there was some work on improving the icons as well.

For a future enhancement for this I'd propose having the ability to dynamically change the Title / Safe Areas adding some sliders to the display panel for this. I originally wanted a scalable system so people could add their custom composition overlays, which is why I suggested the XML file as a sort of template, however the only feasible way would be to load the compositional guides from external SVG's and that might prove just too much for a simple feature.

If anything once SVG import is implemented it should become easier to consider that or even to implement a "guide layer" type that would work for more things, not just for compositional overlays.

I'd say this works for now as a default and any custom shapes should be imported as bitmaps for now, and later as SVG's using the vector layer to simplify things.

Either way good job everyone!
